### PR TITLE
Add `SkillEntry.from_directory`

### PR DESCRIPTION
**Pull request by NeonDaniel**
Add local parsing option with unit tests
Related bugfixes in SkillEntry to ensure property types

### CODE REVIEW

File: ovos_skills_manager/skill_entry.py [hunk 1]
```diff
@@ -100,10 +101,21 @@ def from_github_url(url, branch:str=None, parse_github:bool=True):
...
 
+    @staticmethod
+    def from_directory(skill_dir: str):
+        skill_dir = expanduser(skill_dir)
+        if not isdir(skill_dir):
+            raise ValueError(f"{skill_dir} is not a valid directory")
+
+        data = get_skill_data_from_directory(skill_dir)
+        parse_python_dependencies(data["requirements"].get("python"),
+                                  requests.headers.get("Authorization"))
```
Comment (JarbasAl) [inline ovos_skills_manager/skill_entry.py:112]: not sure this auth should be coming from session...

session can be cleared at any time, usually after each request. and session often changes between appstores so token may not be set

Comment (JarbasAl) [reply ovos_skills_manager/skill_entry.py:112]: can it be made a parameter for the method? i know its not ideal, but the method should be generic enough for this usage

Comment (NeonDaniel) [reply ovos_skills_manager/skill_entry.py:112]: Makes sense; I copy/pasted this out of somewhere else. This only applies to python dependencies that are private git repos, so should be an edge case

**Review by JarbasAl — APPROVED** [review]
@ChanceNCounter any feedback?

**Comment by ChanceNCounter** [comment]
> @ChanceNCounter any feedback?

Not on a 35-file PR I can't personally test, nope.

**Comment by JarbasAl** [comment]
you know the files are basically just test skills right?
